index.js

```diff
@@ -271,7 +282,37 @@ function logError(context, error){
             return null;
         }
       } while (nbResults === pageSize);
-  
+
+    if (version >= "8.0" && !data.noSecurityHotspot) {
+      do {
+        try {
+            const response = await got(`${sonarBaseURL}/api/hotspots/search?projectKey=${data.projectName}&ps=${pageSize}&p=${page}&statuses=TO_REVIEW`, {
+                agent,
+                headers
+            });
+            page++;
+            const json = JSON.parse(response.body);
+            nbResults = json.hotspots.length;
+            data.issues = data.issues.concat(json.hotspots.map(hotspot => {
+              return {
+                rule: undefined,
```

Comment:
is it not possible to match the rule the same way we do with issues?

Reply:
Not really, because there is no "rule" property for a hotspot. API returns something like that:
`{
  "paging": {
    "pageIndex": 1,
    "pageSize": 100,
    "total": 3
  },
  "hotspots": [
    {
      "key": "hotspot-0",
      "component": "com.sonarsource:test-project:src/main/java/com/sonarsource/FourthClass.java",
      "project": "com.sonarsource:test-project",
      "securityCategory": "others",
      "vulnerabilityProbability": "LOW",
      "status": "TO_REVIEW",
      "line": 10,
      "message": "message-0",
      "assignee": "assignee-uuid",
      "author": "joe",
      "creationDate": "2020-01-02T15:43:10+0100",
      "updateDate": "2020-01-02T15:43:10+0100"
    },
    {
      "key": "hotspot-1",
      "component": "com.sonarsource:test-project:src/main/java/com/sonarsource/FourthClass.java",
      "project": "com.sonarsource:test-project",
      "securityCategory": "others",
      "vulnerabilityProbability": "LOW",
      "status": "TO_REVIEW",
      "line": 11,
      "message": "message-1",
      "assignee": "assignee-uuid",
      "author": "joe",
      "creationDate": "2020-01-02T15:43:10+0100",
      "updateDate": "2020-01-02T15:43:10+0100"
    },
    {
      "key": "hotspot-2",
      "component": "com.sonarsource:test-project:src/main/java/com/sonarsource/FourthClass.java",
      "project": "com.sonarsource:test-project",
      "securityCategory": "others",
      "vulnerabilityProbability": "LOW",
      "status": "TO_REVIEW",
      "line": 12,
      "message": "message-2",
      "assignee": "assignee-uuid",
      "author": "joe",
      "creationDate": "2020-01-02T15:43:10+0100",
      "updateDate": "2020-01-02T15:43:10+0100"
    }
  ],
  "components": [
    {
      "key": "com.sonarsource:test-project:src/main/java/com/sonarsource/FourthClass.java",
      "qualifier": "FIL",
      "name": "FourthClass.java",
      "longName": "src/main/java/com/sonarsource/FourthClass.java",
      "path": "src/main/java/com/sonarsource/FourthClass.java"
    },
    {
      "key": "com.sonarsource:test-project",
      "qualifier": "TRK",
      "name": "test-project",
      "longName": "test-project"
    }
  ]
}`

Reply:
it's odd indeed that they didn't keep the rule from the `api/hotspots/search` result. it's there on the `api/hotspots/show` result but it would be a hassle to get fetch them. I think we can leave this as is for now.